src/event_imp.rs

```diff
@@ -981,6 +986,450 @@ fn test_debug_ready() {
+const PRI: usize = 0b100_0000;
+
+/// Interests used in registering.
+///
+/// Interests are used in registering [`Evented`] handles with [`Poll`],
+/// they indicate what readiness should be monitored for. For example if a
+/// socket is registered with readable interests and the socket becomes
+/// writable, no event will be returned from [`poll`].
+///
+/// [`Poll`]: struct.Poll.html                                                 
+/// [`readable`]: #method.readable
+/// [`writable`]: #method.writable
+#[derive(Copy, PartialEq, Eq, Clone, PartialOrd, Ord)]
+pub struct Interests(usize);
+
+impl Interests {
+    /// Returns `Interests` representing readable readiness.
+    ///
+    /// See [`Poll`] for more documentation on polling.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mio::Interests;
+    ///
+    /// let interest = Interests::readable();
+    ///
+    /// assert!(interest.is_readable());
+    /// ```
+    ///
+    /// [`Poll`]: struct.Poll.html
+    #[inline]
+    pub fn readable() -> Interests {
+        Interests(READABLE)
+    }
+
+    /// Returns `Interests` representing writable readiness.
+    ///
+    /// See [`Poll`] for more documentation on polling.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mio::Interests;
+    ///
+    /// let interest = Interests::writable();
+    ///
+    /// assert!(interest.is_writable());
+    /// ```
+    ///
+    /// [`Poll`]: struct.Poll.html
+    #[inline]
+    pub fn writable() -> Interests {
+        Interests(WRITABLE)
+    }
+
+    /// Returns `Interests` representing readable and writable readiness.
+    ///
+    /// See [`Poll`] for more documentation on polling.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mio::Interests;
+    ///
+    /// let interest = Interests::both();
+    ///
+    /// assert!(interest.is_readable());
+    /// assert!(interest.is_writable());
+    /// ```
+    ///
+    /// [`Poll`]: struct.Poll.html
+    #[inline]
+    pub fn both() -> Interests {
+        Interests(READABLE | WRITABLE)
+    }
+
+    /// Returns `Interests` representing HUP readiness.
+    ///
+    /// A HUP (or hang-up) signifies that a stream socket **peer** closed the
+    /// connection, or shut down the writing half of the connection.
+    ///
+    /// **Note that only readable and writable readiness is guaranteed to be
+    /// supported on all platforms**. This means that `hup` readiness
+    /// should be treated as a hint. For more details, see [readiness] in the
+    /// poll documentation.
+    ///
+    /// See [`Poll`] for more documentation on polling.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mio::Interests;
+    ///
+    /// let interest = Interests::hup();
+    ///
+    /// assert!(interest.is_hup());
+    /// ```
+    ///
+    /// [`Poll`]: struct.Poll.html
+    /// [readiness]: ../struct.Poll.html#readiness-operations
+    #[inline]
+    #[cfg(unix)]
+    pub fn hup() -> Interests {
```

Comment:
I don't believe hup interests doesn't make sense as this is registered automatically.

Reply:
Roger